client/landing/stepper/declarative-flow/newsletter.ts

```diff
@@ -71,10 +70,11 @@ const newsletter: Flow = {
 		return [ ...publicSteps, ...privateSteps ];
 	},
 	useSideEffect() {
-		const { setHidePlansFeatureComparison } = useDispatch( ONBOARD_STORE );
+		const { setHidePlansFeatureComparison, setIntent } = useDispatch( ONBOARD_STORE );
 		useEffect( () => {
 			setHidePlansFeatureComparison( true );
 			clearSignupDestinationCookie();
+			setIntent( Onboard.SiteIntent.Newsletter );
```

Comment:
@alshakero Are there any concerns with doing this here, for this flow?

Reply:
No that's looks great.